File: CS/DXSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraBars.Docking;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Registrator;

namespace DXSample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            for (int i = 0; i < imageCollection1.Images.Count; i++) {
                DropDownItem item = new DropDownItem();
                item.ImageIndex = i;
                item.Description = imageCollection1.Images.Keys[i];
                dropDownItemBindingSource.Add(item);
            }
            dataSourceClassBindingSource.AddNew();
            dataSourceClassBindingSource.AddNew();
            dataSourceClassBindingSource.AddNew();
        }
    }
    public class DataSourceClass {
        private string _Name;
        public string Name {
            get { return _Name; }
            set {
                _Name = value;
            }
        }
        private string _ImageName;
        public string ImageName {
            get { return _ImageName; }
            set {
                _ImageName = value;
            }
        }
        
    }
    public class DropDownItem {
        private string _Description;
        public string Description {
            get { return _Description; }
            set {
                _Description = value;
            }
        }
        private int _ImageIndex;
        public int ImageIndex {
            get { return _ImageIndex; }
            set {
                _ImageIndex = value;
            }
        }
        
    }
}

File: VB/DXSample/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace DXSample

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            For i As Integer = 0 To imageCollection1.Images.Count - 1
                Dim item As DropDownItem = New DropDownItem()
                item.ImageIndex = i
                item.Description = imageCollection1.Images.Keys(i)
                dropDownItemBindingSource.Add(item)
            Next

            dataSourceClassBindingSource.AddNew()
            dataSourceClassBindingSource.AddNew()
            dataSourceClassBindingSource.AddNew()
        End Sub
    End Class

    Public Class DataSourceClass

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                _Name = value
            End Set
        End Property

        Private _ImageName As String

        Public Property ImageName As String
            Get
                Return _ImageName
            End Get

            Set(ByVal value As String)
                _ImageName = value
            End Set
        End Property
    End Class

    Public Class DropDownItem

        Private _Description As String

        Public Property Description As String
            Get
                Return _Description
            End Get

            Set(ByVal value As String)
                _Description = value
            End Set
        End Property

        Private _ImageIndex As Integer

        Public Property ImageIndex As Integer
            Get
                Return _ImageIndex
            End Get

            Set(ByVal value As Integer)
                _ImageIndex = value
            End Set
        End Property
    End Class
End Namespace
